src/convnets/inception.jl

```diff
@@ -176,21 +174,419 @@ See also [`inception3`](#).
+                 mixed_5b(),
+                 [block35(0.17f0) for _ in 1:10]...,
+                 mixed_6a(),
+                 [block17(0.10f0) for _ in 1:20]...,
+                 mixed_7a(),
+                 [block8(0.20f0) for _ in 1:9]...,
+                 block8(; activation = relu),
+                 conv_bn((1, 1), 2080, 1536)...)
+    head = Chain(GlobalMeanPool(), MLUtils.flatten, Dropout(dropout), Dense(1536, nclasses))
+    return Chain(body, head)
+end
+
+"""
+    InceptionResNetv2(; pretrain = false, inchannels = 3, dropout = 0.0, nclasses = 1000)
+
+Creates an InceptionResNetv2 model.
+([reference](https://arxiv.org/abs/1602.07261))
+
+# Arguments
+
+  - `pretrain`: set to `true` to load the pre-trained weights for ImageNet
+  - inchannels: number of input channels.
+  - dropout: rate of dropout in classifier head.
+  - nclasses: the number of output classes.
+
+!!! warning
+    
+        `InceptionResNetv2` does not currently support pretrained weights.
+"""
+struct InceptionResNetv2
+    layers::Any
+end
+
+function InceptionResNetv2(; pretrain = false, inchannels = 3, dropout = 0.0,
+                           nclasses = 1000)
+    layers = inceptionresnetv2(; inchannels, dropout, nclasses)
+    pretrain && loadpretrain!(layers, "InceptionResNetv2")
+    return InceptionResNetv2(layers)
+end
+
+@functor InceptionResNetv2
+
+(m::InceptionResNetv2)(x) = m.layers(x)
+
+backbone(m::InceptionResNetv2) = m.layers[1]
+classifier(m::InceptionResNetv2) = m.layers[2]
+
+## Xception
+
+"""
+    xception_block(inchannels, outchannels, nrepeats; stride = 1, start_with_relu = true, 
+                        grow_first = true)
+
+Create an Xception block.
+([reference](https://arxiv.org/abs/1610.02357))
+
+# Arguments
+
+  - inchannels: number of input channels.
+  - outchannels: number of output channels.
+  - nrepeats: number of repeats of depthwise separable convolution layers.
+  - stride: stride by which to downsample the input.
+  - start_with_relu: if true, start the block with a ReLU activation.
+  - grow_first: if true, increase the number of channels at the first convolution.
+"""
+function xception_block(inchannels, outchannels, nrepeats; stride = 1,
+                        start_with_relu = true,
+                        grow_first = true)
+    if outchannels != inchannels || stride != 1
+        skip = conv_bn((1, 1), inchannels, outchannels, identity; stride = stride,
+                       bias = false)
+    else
+        skip = [identity]
+    end
+    layers = []
+    for i in 1:nrepeats
+        if grow_first
+            inc = i == 1 ? inchannels : outchannels
+            outc = outchannels
+        else
+            inc = inchannels
+            outc = i == nrepeats ? outchannels : inchannels
+        end
+        push!(layers, x -> relu.(x))
+        append!(layers,
+                depthwise_sep_conv_bn((3, 3), inc, outc; pad = 1, bias = false,
+                                      use_bn = (false, false)))
```

Comment:
Is there a use case of `(true, false)` or `(false, true)`? If not, I was thinking it is kinda silly to use `conv_bn` with the keyword `use_bn = false` cause that's just `Conv`. It makes sense for consistency with `depthwise_sep_conv_bn`. But it might be cleaner to just introduce `depthwise_sep_conv` then instead of more keywords in multiple places?

Reply:
I can't say I see something _yet_ but I think this can be kept for some time...with all the churn over the next two months it might be handy. If not, I'll simplify it later